api/routers/zip.py

```diff
@@ -0,0 +1,257 @@
+        files_map = _extract_zip(archive_path, extract_dir)
+    except HTTPException:
+        shutil.rmtree(base_dir, ignore_errors=True)
+        raise
+    except zipfile.BadZipFile:
+        shutil.rmtree(base_dir, ignore_errors=True)
+        raise HTTPException(status_code=400, detail="Invalid ZIP file")
+    finally:
+        archive_path.unlink(missing_ok=True)
+
+    if not files_map:
+        shutil.rmtree(base_dir, ignore_errors=True)
+        raise HTTPException(status_code=400, detail="ZIP archive contains no files")
+
+    valid_files = [name for name in files_map if _is_supported(name)]
+    if not valid_files:
+        shutil.rmtree(base_dir, ignore_errors=True)
+        raise HTTPException(
+            status_code=400,
+            detail="ZIP archive must include CSV, CSV.GZ, or Parquet files",
+        )
+
+    zip_id = str(uuid.uuid4())
+    dataset_hint = _sanitize_dataset_name(Path(original_name).stem) or f"zip_{zip_id[:8]}"
+
+    _store_session(
+        zip_id,
+        {
+            "base_dir": str(base_dir),
+            "files": files_map,
+            "source_name": original_name,
+            "suggested_dataset": dataset_hint,
+            "created_at": time.time(),
+        },
+    )
+
+    return {"zip_id": zip_id, "files": sorted(files_map.keys())}
+
+
+@router.post("/ingest_zip_files")
+def ingest_zip_files(request: ZipIngestRequest):
+    session = _get_session(request.zip_id)
+    selected = request.selected_files
+
+    if not selected:
+        raise HTTPException(status_code=400, detail="At least one file must be selected")
+
+    available_files: Dict[str, str] = session["files"]
+    missing = [name for name in selected if name not in available_files]
+    if missing:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Files not found in ZIP: {', '.join(missing)}",
+        )
+
+    invalid = [name for name in selected if not _is_supported(name)]
+    if invalid:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Unsupported file types selected: {', '.join(invalid)}",
+        )
+
+    file_paths = []
+    empty_files = []
+    for name in selected:
+        path = Path(available_files[name])
+        if not path.exists():
+            raise HTTPException(status_code=400, detail=f"File missing on server: {name}")
+        if path.stat().st_size == 0:
+            empty_files.append(name)
+        file_paths.append(str(path))
+
+    if empty_files:
+        raise HTTPException(
+            status_code=400,
+            detail=f"Selected files are empty: {', '.join(empty_files)}",
+        )
+
+    dataset_label = request.dataset_name or session["suggested_dataset"]
+    dataset_id = _sanitize_dataset_name(dataset_label)
+    if not dataset_id:
+        dataset_id = f"zip_{request.zip_id[:8]}"
+
+    source_label = f"zip:{session['source_name']}::{','.join(selected)}"
+
+    try:
+        table_name, n_rows, n_cols = ingest_combined_files(
```

Comment:
I think another button should be added to the frontend for "ingest and combine selected files". we should also keep "ingest selected files" but instead of calling ingest_combined_files() call ingest_file() for every file path. We should be able to do this by just adding a binary flag to the request body.